### DISCUSSION TITLE
OnInitializedAsync getting called twice?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3983] [General] [answered]
for some reason I assumed that OnInitializedAsync for a module action would only get called once, but in debugging I noticed it was getting called twice.  After moving my code to OnParametersSetAsync I noticed it was still getting called twice.  Do I need to move this code to OnAfterRenderAsync and check the firstRender? 

The only things that's special about this action is that it's getting called via a call-back from stripe.  It's the order fulfillment page.  The only interesting thing is passing a parameter from the querystring.

```
 [SupplyParameterFromQuery]
 [Parameter]
 public string session_id { get; set; }
 ```

**Comment by sbwalker** [comment, marked as answer]
@markdav-is OnInitialized is always called twice when Pre-rendering is enabled (which is the default setting) - this is standard Blazor behavior. It is called when the component is initialized on the server, and then again when the component becomes interactive.

**Reply by markdav-is** [reply]
Interesting!  Is there a way to tell and only call once without creating a local flag?  I can make a local flag.  I don't think I can turn off pre-rendering just for one module action.  I'm running on 5.0.2 so I don't think I have the option to make it server side only.

**Reply by sbwalker** [reply]
@markdav-is if you inject the SiteService into your component you can check the IsPrerendering property to determine if the component is prerendering or not.
